pages/api/v1/[serverId].ts

```diff
@@ -0,0 +1,64 @@
+import { deleteServer, getServer, updateServer } from '@/services/server.service';
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { method } = req;
+
+  let serverId: number;
+
+  // Speaking of, TODO: AUTHORIZATIONS
+  try {
+    serverId = parseInt(req.query.serverId as string);
+  }
+
+  catch (err: any) {
+    console.error(err);
+    return res.status(400).send({ error: 'Invalid server ID' });
+  }
+
+  if (method === 'GET') {
+    const { data: server, error } = await getServer(serverId);
+
+    if (error) {
+      return res.status(400).send({ error });
+    }
+
+    return res.status(200).send(server);
+  }
+
+  else if (method === 'PUT') {
+    const { data: server, error } = await updateServer(
+      serverId,
+      req.body.name,
+      req.body.description || null
+    );
+
+    if (error) {
+      return res.status(400).send({ error });
+    }
+
+    return res.status(200).send(server);
+  }
+
+  else if (method === 'DELETE') {
+    // NOTE: auth header cannot be null here as it passes through the auth middleware
+    const decodedJwt = jwt.decode(req.headers.authorization!.replace('Bearer ', '')) as JwtPayload;
+
+    const { data: server, error } = await deleteServer(
+      decodedJwt.sub!,
+      serverId
+    );
+
+    if (error) {
+      return res.status(400).send({ error });
+    }
+
+    return res.status(200).send(server);
```

Comment:
```suggestion
    const {
      data: { user }, error: userError
    } = await supabaseServerClient.auth.getUser();

    if (user) {
      const { data: server, error } = await deleteServer(
        user.id,
        serverId
      );

      if (error) {
        return res.status(400).send({ error });
      }
  
      return res.status(200).send(server);
    }

    if (userError) {
      return res.status(400).send({ userError });
    }

    return res.status(400).json({ message: 'Invalid request' });
```

Reply:
As mentioned in Discord, we should rely on supabase to handle authorization and authentication instead of resolving to manually verfying the tokens. In this case, the ``jwt.decode`` only decodes the token and does not verify it, therefore a faulty jwt may be used. For future reference, ``jwt.verify()`` should ONLY be used when needing to access the jwt, as it accomplishes both decoding the token and verifying it (provided you pass it the signing key from supabase).

Reply:
This was the goal actually

Ideally we have our middleware verifying tokens as we pass through middlware before running the api routes, as such, it would be cleaner to verify the token there as a central point, as all routes will requre the user to at least be signed in

That way, all routes can operate knowing the user is valid and we don't need to check that in every route